### PR TITLE
Replace CopyFileA with a cross-platform function

### PR DESCRIPTION
**Pull request by JanSimek**
See #17

### CODE REVIEW

**Comment by JanSimek** [comment]
Even easier would be to use [std::filesystem::copy_file](https://en.cppreference.com/w/cpp/filesystem/copy_file) but the downside is the dependency on C++17.

**Comment by alexbatalov** [comment]
I guess it's not a bad thing if it allows to move forward. I've checked `filesystem` availability on major compilers and it looks good.

**Comment by alexbatalov** [comment]
I've restored `windows.h` in `art.h` and `audio_file.cc` since it's needed for `MAX_PATH`, which is to be addressed separately.

**Comment by k3tamina** [comment]
Beware of bFailIfExists parameter:
If this parameter is TRUE and the new file specified by lpNewFileName already exists, the function fails. **If this parameter is FALSE and the new file already exists, the function overwrites the existing file and succeeds.**

File: src/file_utils.cc [hunk 1]
```diff
@@ -142,8 +143,6 @@ int _gzdecompress_file(const char* existingFilePath, const char* newFilePath)
 
 void fileCopy(const char* existingFilePath, const char* newFilePath)
 {
-    std::ifstream source { existingFilePath, std::ios::binary };
-    std::ofstream destination { newFilePath, std::ios::binary };
-
-    destination << source.rdbuf();
+    std::error_code ec;
+    std::filesystem::copy_file(std::filesystem::path(existingFilePath), std::filesystem::path(newFilePath), ec);
```
Comment (JanSimek) [inline src/file_utils.cc:147]: Not sure if it is needed but the original CopyFileA function overwrote the destination file but the std::filesystem::copy_file will fail without the [std::filesystem::copy_options::overwrite_existing](https://en.cppreference.com/w/cpp/filesystem/copy_options) option

**Comment by alexbatalov** [comment]
Nice catch guys. Added separate param to overwrite or not, but I guess it's not needed, since all callers are in the same file and they all want to overwrite.

**Pull request merged**